### PR TITLE
perf/test: run malformed header test separately

### LINKED ISSUE #2723: perf/test: run malformed header test separately
### Describe the feature

currently we're running

https://github.com/paradigmxyz/reth/blob/352b55dc0593a2a88e14e7e894079160aeb74010/crates/storage/codecs/derive/src/arbitrary.rs#L46-L54

per proptest, which is a bit expensive,

instead we should generate a new test case that only runs this on a single random instance.

## TODO
* generate a separate test case for rlp malformed header check to speed up CI

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by chirag-bgh**
Closes #2723

### CODE REVIEW

**Review by mattsse — CHANGES_REQUESTED** [review]

File: crates/storage/codecs/derive/src/arbitrary.rs [hunk 1]
```diff
@@ -16,17 +16,16 @@ pub fn maybe_generate_tests(args: TokenStream, ast: &DeriveInput) -> TokenStream
 
     let mut traits = vec![];
     let mut roundtrips = vec![];
+    let mut malformed_checks = vec![];
```
Comment (mattsse) [inline crates/storage/codecs/derive/src/arbitrary.rs:19]: this doesn't do anything yet, or am I missing something?

**Review by mattsse — CHANGES_REQUESTED** [review]

**Comment by onbjerg** [comment]
Ping @chirag-bgh

File: crates/storage/codecs/derive/src/arbitrary.rs [hunk 2]
```diff
@@ -79,6 +79,13 @@ pub fn maybe_generate_tests(args: TokenStream, ast: &DeriveInput) -> TokenStream
                         #(#roundtrips)*
                     });
                 }
+
+                #[test]
+                fn malformed_rlp_checks() {
```
Comment (chirag-bgh) [inline crates/storage/codecs/derive/src/arbitrary.rs:84]: I'm not sure if I understand the issue properly. Is this what intended to do?

Comment (mattsse) [reply crates/storage/codecs/derive/src/arbitrary.rs:84]: yes, the entire point of the referenced issue is to move the malformed header check out of the proptest test so they don't run multiple times (slows down CI)

**Review by mattsse — APPROVED** [review]
okay, so turns out creating random/arbitrary instances isn't actually very straight forward... sorry about this

figured it out after spending way too long reading arbitrary docs lol

**Pull request merged**